org.adempiere.base/src/org/compiere/model/MUser.java

```diff
@@ -1084,6 +1084,56 @@ public static MUser get(Properties ctx, String name) {
 		return retValue;
 	}
 	
+	/**
+	 * Returns true if the user is a system user - without same username in other tenants
+	 * @return
+	 */
+	public static boolean isSystemUser(String name) {
+		if (name == null || name.length() == 0) {
+			s_log.warning ("Invalid Name = " + name);
+			return false;
+		}
+		StringBuilder sql = new StringBuilder("SELECT DISTINCT u.AD_User_ID, u.AD_Client_ID ")
+			.append("FROM AD_User u")
+			.append(" INNER JOIN AD_User_Roles ur ON (u.AD_User_ID=ur.AD_User_ID AND ur.IsActive='Y')")
+			.append(" INNER JOIN AD_Role r ON (ur.AD_Role_ID=r.AD_Role_ID AND r.IsActive='Y') ");
+		sql.append("WHERE u.Password IS NOT NULL AND ");		//	#1/2
+		boolean email_login = MSysConfig.getBooleanValue(MSysConfig.USE_EMAIL_FOR_LOGIN, false);
+		if (email_login)
+			sql.append("u.EMail=?");
+		else
+			sql.append("COALESCE(u.LDAPUser,u.Name)=?");
+		sql.append(" AND u.IsActive='Y'")
+			.append(" AND EXISTS (SELECT * FROM AD_Client c WHERE u.AD_Client_ID=c.AD_Client_ID AND c.IsActive='Y')")
+			.append(" ORDER BY u.AD_Client_ID");
```

Comment:
This looks fine for email login but it is inconsistent with other part of the application for user name login. AFAIU, the current authentication model is you can have same user name across tenant and the identify of an user is define by username + password.